docs(core): add docs for parallelism (#26950)
nrwl/nx: docs/shared/reference/project-configuration.md

{% /callout %}

### Parallelism

> In Nx 19.5.0 and higher, you can configure parallelism for a target by setting `"parallelism": false` in the target's configuration. This ensures that the target will not run in parallel with other tasks. For example, if you want the `e2e` target to run tests alone, you can specify:

Reviewer: ```suggestion
In Nx 19.5.0+, you can configure parallelism for a target by setting `"parallelism": false` in the target's configuration. This ensures that the target will not run in parallel with other tasks. For example, if you do not want the `e2e` target to alongside any other tasks, you can specify:
```

```json {% fileName="project.json" %}
{
  "targets": {
    "e2e": {
      "parallelism": false
    }
  }
}
```

> Parallelism is a feature in Nx that allows multiple tasks to run concurrently. By default, Nx tries to execute as many tasks in parallel as possible to optimize build and test times. However, there are scenarios where running tasks in parallel is not desirable. For example, tasks that rely on shared resources or have side effects might fail or produce inconsistent results if run concurrently.

Reviewer: ```suggestion
Parallelism is a feature in Nx that allows multiple tasks to run in parallel. By default, Nx tries to execute tasks in parallel to optimize build and test times. However, there are scenarios where running tasks in parallel is not desirable. For example, tasks that rely on shared resources or have side effects might fail or produce inconsistent results when run alongside other tasks.
```

To control parallelism, you can set the "parallelism" option in a target's configuration. This option is available starting from version 19.5.0 and can be set to either true or false.

- `"parallelism": true` (default): The task can run in parallel with other tasks.

> - `"parallelism": false`: The task will run alone, without any other tasks running concurrently.

Reviewer: Remove
